BacktraceScreen.c

```diff
@@ -0,0 +1,527 @@
+*/
+
+#include "config.h" // IWYU pragma: keep
+
+#include "BacktraceScreen.h"
+
+#if defined(HAVE_BACKTRACE_SCREEN)
+
+#include <assert.h>
+#include <limits.h>
+#include <stdbool.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/wait.h>
+
+#include "CRT.h"
+#include "FunctionBar.h"
+#include "Macros.h"
+#include "Object.h"
+#include "Panel.h"
+#include "Platform.h"
+#include "Process.h"
+#include "ProvideCurses.h"
+#include "RichString.h"
+#include "Settings.h"
+#include "Vector.h"
+#include "XUtils.h"
+
+
+#define MAX_HEX_ADDR_STR_LEN_32 (strlen("0x") + 8)
+#define MAX_HEX_ADDR_STR_LEN_64 (strlen("0x") + 16)
+
+typedef enum BacktracePanelOptions_ {
+   OPTION_NAME_DEMANGLE,
+   OPTION_NAME_RAW,
+   OPTION_OBJECT_FULL_PATH,
+   OPTION_OBJECT_BASENAME,
+   LAST_PANEL_OPTION,
+} BacktracePanelOptions;
+
+static const char* const BacktracePanel_options[LAST_PANEL_OPTION] = {
+   [OPTION_NAME_DEMANGLE] = "Demangle",
+   [OPTION_NAME_RAW] = "Raw",
+   [OPTION_OBJECT_FULL_PATH] = "Full Path",
+   [OPTION_OBJECT_BASENAME] = "Basename",
+};
+
+typedef enum BacktraceFrameHeaders_ {
+   HEADER_NUMBER_FRAME,
+   HEADER_ADDRESS,
+   HEADER_NAME,
+   HEADER_PATH,
+   LAST_PANEL_HEADER,
+} BacktracePanelHeaders;
+
+static const char* const BacktraceFrame_headerFields[LAST_PANEL_HEADER] = {
+   [HEADER_NUMBER_FRAME] = "#",
+   [HEADER_ADDRESS] = "ADDRESS",
+   [HEADER_NAME] = "NAME",
+   [HEADER_PATH] = "PATH",
+};
+
+static const char* const BacktraceScreenFunctions[] = {
+   "Refresh",
+#if defined(HAVE_DEMANGLING)
+   BacktracePanel_options[OPTION_NAME_RAW],
+#endif
+   BacktracePanel_options[OPTION_OBJECT_FULL_PATH],
+   "Done   ",
+   NULL
+};
+
+static const char* const BacktraceScreenKeys[] = {
+   "F1",
+#if defined(HAVE_DEMANGLING)
+   "F2",
+#endif
+   "F3",
+   "Esc",
+   NULL
+};
+
+static const int BacktraceScreenEvents[] = {
+   KEY_F(1),
+#if defined(HAVE_DEMANGLING)
+   KEY_F(2),
+#endif
+   KEY_F(3),
+   27,
+};
+
+typedef enum BacktraceScreenDisplayOptions_ {
+   NO_OPTION = 0,
+   DEMANGLE_NAME_FUNCTION = 1 << 0,
+   SHOW_FULL_PATH_OBJECT = 1 << 1,
+} BacktraceScreenDisplayOptions;
+
+BacktraceFrameData* BacktraceFrameData_new(void) {
+   BacktraceFrameData *this = AllocThis(BacktraceFrameData);
+   this->index = -1;
+   this->address = 0;
+   this->offset = 0;
+   this->functionName = NULL;
+   this->demangleFunctionName = NULL;
+   this->isSignalFrame = false;
+   this->objectPath = NULL;
+   this->objectName = NULL;
+   return this;
+}
+
+void BacktraceFrameData_delete(Object* object) {
+   BacktraceFrameData* this = (BacktraceFrameData*)object;
+   free(this->functionName);
+   free(this->demangleFunctionName);
+   free(this->objectPath);
+   free(this->objectName);
+   free(this);
+}
+
+static void BacktracePanel_displayHeader(BacktracePanel* this) {
+   const BacktracePanelPrintingHelper* printingHelper = &this->printingHelper;
+   const int displayOptions = this->displayOptions;
```

Comment:
```suggestion
   const BacktracePanelPrintingHelper* ph = &this->printingHelper;
   const int opt = this->displayOptions;
```

Shorten the identifiers. Keeps lines short and easy to parse.

Reply:
I cannot rename `printingHelper` to `ph` because its value will never be between 0 and 14 :joy:

> Keeps lines short and easy to parse

For who? The developer or the compiler? I'm not sure about the benefit of a variable with two letters.